test/modules/TestResultCoordinator/report/IResults.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator
+{
+    using System;
+    using System.Threading.Tasks;
+
+    // TODO: Replace IResults with IAsyncEnumerable when project is moved to .Net Core 3.1
+    public interface IResults<T> : IDisposable
+    {
+        T Current { get; }
+
+        Task<bool> MoveNextAsync();
```

Comment:
currently sequentialStore has GetBatchAsync method; which will reduce number of calls to load data from disk.  We should implement GetBatchAsync instead.